level2/asg.tf

```diff
@@ -1,10 +1,80 @@
-module "asg" {
-  source = "../modules/asg"
-
-  env_code          = var.env_code
-  vpc_id            = data.terraform_remote_state.level1.outputs.vpc_id
-  private_subnet_id = data.terraform_remote_state.level1.outputs.private_subnet_id
-  load_balance_sg   = module.lb.load_balancer_sg
-  target_group_arn  = module.lb.target_group_arn
+module "private_sg" {
+  source = "terraform-aws-modules/security-group/aws"
+
+  name        = "${var.env_code}-private"
+  description = "Allow port 80 and 3306 TCP inbound to ec2 ASG instances within VPC"
+  vpc_id      = data.terraform_remote_state.level1.outputs.vpc_id
+
+  computed_ingress_with_source_security_group_id = [
+    {
+      rule                     = "http-80-tcp"
+      source_security_group_id = module.external_sg.security_group_id
+    }
+  ]
+  number_of_computed_ingress_with_source_security_group_id = 1
+
+  egress_with_cidr_blocks = [
+    {
+      from_port   = 0
+      to_port     = 65535
+      protocol    = "tcp"
+      description = "http to ELB"
+      cidr_blocks = "0.0.0.0/0"
+    }
+  ]
+
+}
+
+data "aws_ami" "amazonlinux" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-kernel-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["137112412989"]
+
+}
+
+
+module "autoscaling" {
+  source = "terraform-aws-modules/autoscaling/aws"
+
+  name                      = var.env_code
+  min_size                  = 2
+  max_size                  = 5
+  desired_capacity          = 4
+  health_check_grace_period = 400
+  health_check_type         = "EC2"
+  vpc_zone_identifier       = data.terraform_remote_state.level1.outputs.private_subnet_id
```

Comment:
Add this line to the asg module
`target_group_arns  = module.elb.target_group_arns`